tensorflow/lite/examples/ios/camera/CameraExampleViewController.mm

```diff
@@ -326,7 +326,13 @@ - (void)runModelOnFrame:(CVPixelBufferRef)pixelBuffer {
   NSLog(@"Time: %.4lf, avg: %.4lf, count: %d", end - start, total_latency / total_count,
         total_count);
 
-  const int output_size = 1000;
+  // read output size from the output sensor
+  const int output_tensor_index = interpreter->outputs()[0];
+  TfLiteTensor* output_tensor = interpreter->tensor(output_tensor_index);
+  TfLiteIntArray* output_dims = output_tensor->dims;
+  assert(output_dims->size == 2);
+  const int output_size = output_dims->data[1]-output_dims->data[0];
```

Comment:
Hmm, is the output shape (1, 1000) or (1000, 1)?

Reply:
The output shape is (1,1001).  There are 1000 categories and first label is background for anything not in any category. I am wondering if the original logic was incorrect and the hardcoded value should have been 1001. That is the existing code will always miss the probability of the last label.

Instead of assert, it will be better to do the following check.
if(output_dims->size != 2 || output_dims->data[0] != 1) {
   LOG(FATAL) << "Output of the model is in invalid format.";
} 
const int output_size = output_dims->data[1];

Reply:
Thanks for your comment @shashishekhar. Fixed in https://github.com/jakublipinski/tensorflow/commit/ab65c34bc6348828d7c682fdc4ad93bb1463eb2f according to your suggestion.